app/decorators/guest_decorator.rb

```diff
@@ -0,0 +1,13 @@
+class GuestDecorator
+  def lockout_time_remaining_in_words
+    current_time = Time.zone.now
+
+    distance_of_time_in_words(
+      current_time, current_time + lockout_time_remaining, true, highest_measures: 2
+    )
+  end
+
+  def lockout_time_remaining
+    (Devise.direct_otp_valid_for - (Time.zone.now - Time.zone.now)).to_i
```

Comment:
To clarify, I'm saying that we don't need this `GuestDecorator` class at all. We just need one class that defines the `lockout_time_remaining` and `lockout_time_remaining_in_words` methods, whether it stays in `UserDecorator` or gets moved to a new class.